plotting/postprocessing_utils.py

```diff
@@ -0,0 +1,311 @@
+import os
+import h5py
+import numpy as np
+import ffmpeg
+import matplotlib.pyplot as plt
+from matplotlib import rc as pltFont
+
+from pygyro.initialisation.constants import get_constants
+from pygyro import splines as spl
+from pygyro.tools.getSlice import get_grid_slice, get_flux_surface_grid_slice
+from pygyro.tools.getPhiSlice import get_phi_slice
+from pygyro.arakawa.utilities import compute_int_f, compute_int_f_squared, get_total_energy
+
+
+def unpack_all(foldername):
+    """
+    Unpacks all Poloidal slices from the grid_xxxxxx.h5 and phi_xxxxxx.h5
+    files in the given folder. For v and z we use the default parameters
+    nv//2 and 0, as used in plot_poloidal_slice.py.
+
+    Parameters
+    ----------
+    foldername  : str
+                The folder containing the simulation
+    """
+    filelist = os.listdir(foldername)
+
+    for f in filelist:
+        isgrid = f.find("grid_")
+        if isgrid != -1:
+            f_n = int(f[5:-3])
+            get_grid_slice(foldername, f_n, "Slices_f")
+            get_flux_surface_grid_slice(foldername, f_n, "FluxSlices_f")
+
+        isphi = f.find("phi_")
+        if isphi != -1:
+            phi_n = int(f[4:-3])
+            get_phi_slice(foldername, phi_n, "Slices_phi")
+
+
+def plot_all_slices(foldername):
+    """
+    Given a folder of poloidal slices,
+    we plot all of them in a subfolder called plots/.
+
+    Parameters
+    ----------
+    foldername  : str
+                The folder containing the Poloidal Slices
+
+    """
+
+    if (not os.path.isdir(foldername+"plots/")):
+        os.mkdir(foldername+"plots/")
+
+    filelist = os.listdir(foldername)
+    for fname in filelist:
+        if fname != "plots":
+            filename = foldername+fname
+            t_str = filename[filename.rfind('_')+1:filename.rfind('.')]
+            t = int(t_str)
+
+            file = h5py.File(filename, 'r')
+            dataset = file['/dset']
+
+            shape = dataset.shape
+            data_shape = list(shape)
+            data_shape[0] += 1
+
+            data = np.ndarray(data_shape)
+
+            data[:-1, :] = dataset[:]
+            data[-1, :] = data[0, :]
+
+            file.close()
+            if np.isnan(data).any() or np.isinf(data).any():
+                print("NaN or Inf encounter")
+            else:
+                superfolder = foldername[0:foldername.find('/')]
```

Comment:
It would be safer to use the `os` module here. E.g `os.path.dirname(foldername)`